New cppcheck errors

**Describe the Bug**
A new version of cppcheck brings many new errors:
```
FAIL: test_projects_with_cppcheck (test_cppcheck.TestCppCheck)
Test projects with Cppcheck.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\webots-r0dha\tests\sources\test_cppcheck.py", line 171, in test_projects_with_cppcheck
    self.run_cppcheck(command)
  File "C:\projects\webots-r0dha\tests\sources\test_cppcheck.py", line 57, in run_cppcheck
    msg='Cppcheck detected some errors:\n\n%s' % reportText
AssertionError: False is not true : Cppcheck detected some errors:
projects\robots\gctronic\e-puck\plugins\robot_windows\botstudio\e-puck\EPuckFacade.cpp:148:27: warning: Either the condition 'groundSensorIndex<NUMBER_OF_GROUND_SENSORS' is redundant or the array 'mGroundSensors[3]' is accessed at index 3, which is out of bounds. [arrayIndexOutOfBoundsCond]
        if (mGroundSensors[groundSensorIndex] == 0)
                          ^
projects\robots\gctronic\e-puck\plugins\robot_windows\botstudio\e-puck\EPuckFacade.cpp:143:31: note: Assuming that condition 'groundSensorIndex<NUMBER_OF_GROUND_SENSORS' is not redundant
        if (groundSensorIndex < NUMBER_OF_GROUND_SENSORS && mGroundSensors[groundSensorIndex] == 0) {
                              ^
projects\robots\gctronic\e-puck\plugins\robot_windows\botstudio\e-puck\EPuckFacade.cpp:148:27: note: Array index out of bounds
        if (mGroundSensors[groundSensorIndex] == 0)
                          ^
projects\robots\kondo\khr-2hv\controllers\khr-2hv_demo\rcb3_simulator.cpp:685:19: style: Obsolete function 'strdup' called. It is recommended to use '_strdup' instead. [strdupCalled]
  motion.m_name = strdup(filename);
                  ^
projects\robots\kondo\khr-2hv\controllers\khr-2hv_demo\rcb3_simulator.cpp:755:25: style: Obsolete function 'strdup' called. It is recommended to use '_strdup' instead. [strdupCalled]
          item.m_name = strdup(buffer + 5);
                        ^
projects\robots\mobsya\thymio\controllers\thymio2_demo\thymio2_demo.c:198:85: style:inconclusive: Boolean expression 'buttons_pressed[0]' is used in bitwise operation. Did you mean '||'? [bitwiseOnBoolean]
    wb_led_set(leds_buttons[LED_BUTTON_BACKWARD], (buttons_pressed[BUTTON_BACKWARD] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
                                                                                    ^
projects\robots\mobsya\thymio\controllers\thymio2_demo\thymio2_demo.c:199:83: style:inconclusive: Boolean expression 'buttons_pressed[3]' is used in bitwise operation. Did you mean '||'? [bitwiseOnBoolean]
    wb_led_set(leds_buttons[LED_BUTTON_FORWARD], (buttons_pressed[BUTTON_FORWARD] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
                                                                                  ^
projects\robots\mobsya\thymio\controllers\thymio2_demo\thymio2_demo.c:200:79: style:inconclusive: Boolean expression 'buttons_pressed[4]' is used in bitwise operation. Did you mean '||'? [bitwiseOnBoolean]
    wb_led_set(leds_buttons[LED_BUTTON_RIGHT], (buttons_pressed[BUTTON_RIGHT] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
                                                                              ^
projects\robots\mobsya\thymio\controllers\thymio2_demo\thymio2_demo.c:201:77: style:inconclusive: Boolean expression 'buttons_pressed[1]' is used in bitwise operation. Did you mean '||'? [bitwiseOnBoolean]
    wb_led_set(leds_buttons[LED_BUTTON_LEFT], (buttons_pressed[BUTTON_LEFT] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
                                                                            ^
projects\samples\curriculum\controllers\advanced_slam_2\advanced_slam_2.c:323:11: warning: Either the condition 'robot_x+2>0' is redundant or the array 'om[35][35]' is accessed at index om[-3][*], which is out of bounds. [negativeIndex]
        om[robot_x - 2][robot_y] = (float)1 / (float)19;
          ^
projects\samples\curriculum\controllers\advanced_slam_2\advanced_slam_2.c:322:23: note: Assuming that condition 'robot_x+2>0' is not redundant
      if (robot_x + 2 > 0)
                      ^
projects\samples\curriculum\controllers\advanced_slam_2\advanced_slam_2.c:323:11: note: Negative array index
        om[robot_x - 2][robot_y] = (float)1 / (float)19;
          ^
======================================================================
FAIL: test_sources_with_cppcheck (test_cppcheck.TestCppCheck)
Test Webots with Cppcheck.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\webots-r0dha\tests\sources\test_cppcheck.py", line 136, in test_sources_with_cppcheck
    self.run_cppcheck(command)
  File "C:\projects\webots-r0dha\tests\sources\test_cppcheck.py", line 57, in run_cppcheck
    msg='Cppcheck detected some errors:\n\n%s' % reportText
AssertionError: False is not true : Cppcheck detected some errors:
resources\projects\plugins\robot_windows\generic\generic.c:84:18: style: Obsolete function 'strdup' called. It is recommended to use '_strdup' instead. [strdupCalled]
  char *tokens = strdup(command);
                 ^
resources\projects\plugins\robot_windows\generic\generic.c:135:18: style: Obsolete function 'strdup' called. It is recommended to use '_strdup' instead. [strdupCalled]
  char *tokens = strdup(commands);
                 ^
src\lib\Controller\api\camera.c:482:7: style: Condition 'c' is always true [knownConditionTrueFalse]
  if (c && in_range) {
      ^
src\lib\Controller\api\camera.c:471:7: note: Assuming that condition '!c' is not redundant
  if (!c || !ac) {
      ^
src\lib\Controller\api\camera.c:482:7: note: Condition 'c' is always true
  if (c && in_range) {
      ^
src\lib\Controller\api\robot.c:963:12: style: Obsolete function 'strdup' called. It is recommended to use '_strdup' instead. [strdupCalled]
    pipe = strdup(WEBOTS_SERVER);
           ^
src\lib\Controller\api\robot.c:982:20: style: Obsolete function 'strdup' called. It is recommended to use '_strdup' instead. [strdupCalled]
            pipe = strdup(buffer);
                   ^
src\lib\Controller\api\system.c:80:3: error: Memory leak: w_short_path [memleak]
  return (const char *)buffer;
  ^
src\lib\Controller\api\supervisor.c:1290:9: style: Condition '!filename' is always false [knownConditionTrueFalse]
    if (!filename || !filename[0]) {
        ^
src\lib\Controller\api\supervisor.c:1289:7: note: Assuming that condition 'filename' is not redundant
  if (filename) {
      ^
src\lib\Controller\api\supervisor.c:1290:9: note: Condition '!filename' is always false
    if (!filename || !filename[0]) {
        ^
src\webots\vrml\WbProtoList.hpp:70:3: error: There is an unknown macro here somewhere. Configuration is required. If Q_DISABLE_COPY is a macro then please configure it. [unknownMacro]
  Q_DISABLE_COPY(WbProtoList)
  ^
src\webots\vrml\WbProtoModel.hpp:109:3: error: There is an unknown macro here somewhere. Configuration is required. If Q_DISABLE_COPY is a macro then please configure it. [unknownMacro]
  Q_DISABLE_COPY(WbProtoModel)
  ^
src\webots\nodes\WbBallJoint.cpp:634:25: warning:inconclusive: Possible null pointer dereference: motor2() [nullPointer]
  if (motor2() && motor2()->isPIDPositionControl())
                        ^
src\webots\nodes\WbBallJoint.cpp:642:25: warning:inconclusive: Possible null pointer dereference: motor3() [nullPointer]
  if (motor3() && motor3()->isPIDPositionControl())
                        ^
src\webots\nodes\WbBaseNode.cpp:130:19: warning:inconclusive: Possible null pointer dereference: boundingSphere() [nullPointer]
    boundingSphere()->resetGlobalCoordinatesUpdateTime();
                  ^
src\webots\nodes\utils\WbNodeOperations.cpp:365:15: warning:inconclusive: Either the condition 'sfnode||mfnode' is redundant or there is possible null pointer dereference: mfnode. [nullPointerRedundantCheck]
    success = mfnode->removeNode(node);
              ^
src\webots\nodes\utils\WbNodeOperations.cpp:358:17: note: Assuming that condition 'sfnode||mfnode' is not redundant
  assert(sfnode || mfnode);
                ^
src\webots\nodes\utils\WbNodeOperations.cpp:357:46: note: Assignment 'mfnode=dynamic_cast<WbMFNode*>(parentField->value())', assigned value is 0
  WbMFNode *mfnode = dynamic_cast<WbMFNode *>(parentField->value());
                                             ^
src\webots\nodes\utils\WbNodeOperations.cpp:365:15: note: Null pointer dereference
    success = mfnode->removeNode(node);
              ^
src\wren\StaticMesh.cpp:537:19: style: Variable 'hash' is assigned a value that is never used. [unreadVariable]
    uint64_t hash = cache::sipHash13c(reinterpret_cast<const char *>(¶ms[0]), params.size() * sizeof(float));
                  ^
src\wren\StaticMesh.cpp:1481:19: style: Variable 'hash' is assigned a value that is never used. [unreadVariable]
    uint64_t hash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
                  ^
src\wren\StaticMesh.cpp:1524:19: style: Variable 'hash' is assigned a value that is never used. [unreadVariable]
    uint64_t hash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
                  ^
src\wren\StaticMesh.cpp:1589:12: style: Variable 'hash' is assigned a value that is never used. [unreadVariable]
      hash = ~hash;
           ^
```
https://ci.appveyor.com/project/omichel/webots-r0dha#L364

--- a/projects/robots/gctronic/e-puck/plugins/robot_windows/botstudio/e-puck/EPuckFacade.cpp
+++ b/projects/robots/gctronic/e-puck/plugins/robot_windows/botstudio/e-puck/EPuckFacade.cpp
@@ -143,9 +143,7 @@ void EPuckFacade::initDevices() {
         if (groundSensorIndex < NUMBER_OF_GROUND_SENSORS && mGroundSensors[groundSensorIndex] == 0) {
           mGroundSensors[groundSensorIndex] = tag;
           gGroundSensorsExist = true;
-        }
-
-        if (mGroundSensors[groundSensorIndex] == 0)
+        } else
           printDeviceError(deviceName);
       }
     }

--- a/projects/robots/mobsya/thymio/controllers/thymio2_demo/thymio2_demo.c
+++ b/projects/robots/mobsya/thymio/controllers/thymio2_demo/thymio2_demo.c
@@ -195,10 +195,10 @@ int main(int argc, char **argv) {
     buttons_pressed[BUTTON_RIGHT] |= (wb_touch_sensor_get_value(buttons[BUTTON_RIGHT]) == 1);
     buttons_pressed[BUTTON_LEFT] |= (wb_touch_sensor_get_value(buttons[BUTTON_LEFT]) == 1);
 
-    wb_led_set(leds_buttons[LED_BUTTON_BACKWARD], (buttons_pressed[BUTTON_BACKWARD] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
-    wb_led_set(leds_buttons[LED_BUTTON_FORWARD], (buttons_pressed[BUTTON_FORWARD] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
-    wb_led_set(leds_buttons[LED_BUTTON_RIGHT], (buttons_pressed[BUTTON_RIGHT] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
-    wb_led_set(leds_buttons[LED_BUTTON_LEFT], (buttons_pressed[BUTTON_LEFT] | buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
+    wb_led_set(leds_buttons[LED_BUTTON_BACKWARD], (buttons_pressed[BUTTON_BACKWARD] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
+    wb_led_set(leds_buttons[LED_BUTTON_FORWARD], (buttons_pressed[BUTTON_FORWARD] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
+    wb_led_set(leds_buttons[LED_BUTTON_RIGHT], (buttons_pressed[BUTTON_RIGHT] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
+    wb_led_set(leds_buttons[LED_BUTTON_LEFT], (buttons_pressed[BUTTON_LEFT] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
 
     if (buttons_pressed[BUTTON_CENTER]) {
       const double *acc_values = wb_accelerometer_get_values(acc);

--- a/projects/robots/mobsya/thymio/controllers/thymio2_demo/thymio2_demo.c
+++ b/projects/robots/mobsya/thymio/controllers/thymio2_demo/thymio2_demo.c
@@ -195,7 +195,8 @@ int main(int argc, char **argv) {
     buttons_pressed[BUTTON_RIGHT] |= (wb_touch_sensor_get_value(buttons[BUTTON_RIGHT]) == 1);
     buttons_pressed[BUTTON_LEFT] |= (wb_touch_sensor_get_value(buttons[BUTTON_LEFT]) == 1);
 
-    wb_led_set(leds_buttons[LED_BUTTON_BACKWARD], (buttons_pressed[BUTTON_BACKWARD] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
+    wb_led_set(leds_buttons[LED_BUTTON_BACKWARD],
+               (buttons_pressed[BUTTON_BACKWARD] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
     wb_led_set(leds_buttons[LED_BUTTON_FORWARD], (buttons_pressed[BUTTON_FORWARD] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
     wb_led_set(leds_buttons[LED_BUTTON_RIGHT], (buttons_pressed[BUTTON_RIGHT] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);
     wb_led_set(leds_buttons[LED_BUTTON_LEFT], (buttons_pressed[BUTTON_LEFT] || buttons_pressed[BUTTON_CENTER]) ? 32 : 0);

--- a/projects/samples/curriculum/controllers/advanced_slam_2/advanced_slam_2.c
+++ b/projects/samples/curriculum/controllers/advanced_slam_2/advanced_slam_2.c
@@ -317,9 +317,9 @@ int main(int argc, char **argv) {
         om[robot_x][robot_y + 1] = (float)2 / (float)19;
       if (robot_y - 1 > 0)
         om[robot_x][robot_y - 1] = (float)2 / (float)19;
-      if (robot_x - 2 < MAP_SIZE)
+      if (robot_x + 2 < MAP_SIZE)
         om[robot_x + 2][robot_y] = (float)1 / (float)19;
-      if (robot_x + 2 > 0)
+      if (robot_x - 2 > 0)
         om[robot_x - 2][robot_y] = (float)1 / (float)19;
       if (robot_y + 2 < MAP_SIZE)
         om[robot_x][robot_y + 2] = (float)1 / (float)19;

--- a/src/lib/Controller/api/camera.c
+++ b/src/lib/Controller/api/camera.c
@@ -479,7 +479,7 @@ void wb_camera_set_focal_distance(WbDeviceTag tag, double focal_distance) {
             c->max_focal_distance);
     in_range = false;
   }
-  if (c && in_range) {
+  if (in_range) {
     c->focal_distance = focal_distance;
     c->set_focal_distance = true;
   }

--- a/src/lib/Controller/api/supervisor.c
+++ b/src/lib/Controller/api/supervisor.c
@@ -1287,8 +1287,8 @@ bool wb_supervisor_world_save(const char *filename) {
     return false;
 
   if (filename) {
-    if (!filename || !filename[0]) {
-      fprintf(stderr, "Error: wb_supervisor_world_save() called with NULL or empty 'filename' argument.\n");
+    if (!filename[0]) {
+      fprintf(stderr, "Error: wb_supervisor_world_save() called with an empty 'filename' argument.\n");
       return false;
     }
 
@@ -1296,6 +1296,9 @@ bool wb_supervisor_world_save(const char *filename) {
       fprintf(stderr, "Error: the target file given to wb_supervisor_world_save() ends with the '.wbt' extension.\n");
       return false;
     }
+  } else {
+    fprintf(stderr, "Error: wb_supervisor_world_save() called with a NULL 'filename' argument.\n");
+    return false;
   }
 
   free(save_filename);

--- a/src/lib/Controller/api/system.c
+++ b/src/lib/Controller/api/system.c
@@ -77,6 +77,7 @@ const char *wbu_system_short_path(const char *path) {
     free(buffer);
   buffer = (char *)malloc(size);
   WideCharToMultiByte(CP_UTF8, 0, w_short_path, -1, buffer, size, NULL, NULL);
+  // cppcheck-suppress memleak
   return (const char *)buffer;
 #else
   return path;

--- a/src/webots/nodes/WbBallJoint.cpp
+++ b/src/webots/nodes/WbBallJoint.cpp
@@ -620,7 +620,7 @@ void WbBallJoint::prePhysicsStep(double ms) {
 
 void WbBallJoint::postPhysicsStep() {
   assert(mJoint);
-  WbMotor *m1 = motor();
+  const WbMotor *const m1 = motor();
   if (m1 && m1->isPIDPositionControl())
     // if controlling in position we update position using directly the angle feedback
     mPosition = WbMathsUtilities::normalizeAngle(-dJointGetAMotorAngle(mControlMotor, 0) + mOdePositionOffset, mPosition);

--- a/src/webots/nodes/WbBallJoint.cpp
+++ b/src/webots/nodes/WbBallJoint.cpp
@@ -620,7 +620,8 @@ void WbBallJoint::prePhysicsStep(double ms) {
 
 void WbBallJoint::postPhysicsStep() {
   assert(mJoint);
-  if (motor() && motor()->isPIDPositionControl())
+  WbMotor *m1 = motor();
+  if (m1 && m1->isPIDPositionControl())
     // if controlling in position we update position using directly the angle feedback
     mPosition = WbMathsUtilities::normalizeAngle(-dJointGetAMotorAngle(mControlMotor, 0) + mOdePositionOffset, mPosition);
   else
@@ -631,7 +632,8 @@ void WbBallJoint::postPhysicsStep() {
   if (p)
     p->setPositionFromOde(mPosition);
 
-  if (motor2() && motor2()->isPIDPositionControl())
+  const WbMotor *const m2 = motor2();
+  if (m2 && m2->isPIDPositionControl())
     mPosition2 = WbMathsUtilities::normalizeAngle(-dJointGetAMotorAngle(mControlMotor, 1) + mOdePositionOffset2, mPosition2);
   else
     mPosition2 -= dJointGetAMotorAngle(mControlMotor, 1) * mTimeStep / 1000.0;
@@ -639,7 +641,8 @@ void WbBallJoint::postPhysicsStep() {
   if (p2)
     p2->setPositionFromOde(mPosition2);
 
-  if (motor3() && motor3()->isPIDPositionControl())
+  const WbMotor *const m3 = motor3();
+  if (m3 && m3->isPIDPositionControl())
     mPosition3 = WbMathsUtilities::normalizeAngle(-dJointGetAMotorAngle(mControlMotor, 2) + mOdePositionOffset3, mPosition3);
   else
     mPosition3 -= dJointGetAMotorAngle(mControlMotor, 2) * mTimeStep / 1000.0;

--- a/src/webots/nodes/WbBaseNode.cpp
+++ b/src/webots/nodes/WbBaseNode.cpp
@@ -126,8 +126,9 @@ void WbBaseNode::validateProtoNodes() {
 
 void WbBaseNode::reset() {
   WbNode::reset();
-  if (boundingSphere())
-    boundingSphere()->resetGlobalCoordinatesUpdateTime();
+  const WbBoundingSphere *const nodeBoundingSphere = boundingSphere();
+  if (nodeBoundingSphere)
+    nodeBoundingSphere->resetGlobalCoordinatesUpdateTime();
 }
 
 //////////////////////////

--- a/src/webots/nodes/utils/WbNodeOperations.cpp
+++ b/src/webots/nodes/utils/WbNodeOperations.cpp
@@ -362,6 +362,7 @@ bool WbNodeOperations::deleteNode(WbNode *node, bool fromSupervisor) {
     sfnode->setValue(NULL);
     success = true;
   } else {
+    assert(mfnode);
     success = mfnode->removeNode(node);
     delete node;
   }

--- a/src/webots/vrml/WbProtoList.hpp
+++ b/src/webots/vrml/WbProtoList.hpp
@@ -67,6 +67,7 @@ public:
   void readModel(WbTokenizer *tokenizer, const QString &worldPath);
 
 private:
+  // cppcheck-suppress unknownMacro
   Q_DISABLE_COPY(WbProtoList)
 
   QString mPrimarySearchPath;

--- a/src/webots/vrml/WbProtoModel.hpp
+++ b/src/webots/vrml/WbProtoModel.hpp
@@ -106,6 +106,7 @@ public:
   WbNode *generateRoot(const QVector<WbField *> &parameters, const QString &worldPath, int uniqueId = -1);
 
 private:
+  // cppcheck-suppress unknownMacro
   Q_DISABLE_COPY(WbProtoModel)
 
   QMap<QString, QString> mStaticContentMap;

--- a/src/wren/StaticMesh.cpp
+++ b/src/wren/StaticMesh.cpp
@@ -534,11 +534,11 @@ namespace wren {
 
     const std::array<float, 3> params = {{static_cast<float>(dimensionX), static_cast<float>(dimensionZ), thickness2}};
 
-    uint64_t hash = cache::sipHash13c(reinterpret_cast<const char *>(&params[0]), params.size() * sizeof(float));
+    uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(&params[0]), params.size() * sizeof(float));
     // cppcheck-suppress uninitvar
-    hash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(heightData)),
+    meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(heightData)),
                               sizeof(float) * dimensionX * dimensionZ);
-    const cache::Key key(hash);
+    const cache::Key key(meshHash);
 
     StaticMesh *mesh;
     if (StaticMesh::createOrRetrieveFromCache(&mesh, key))
@@ -1478,16 +1478,16 @@ namespace wren {
   }
 
   StaticMesh *StaticMesh::createLineSet(int coordCount, const float *coordData, const float *colorData) {
-    uint64_t hash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
+    uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
                                       sizeof(glm::vec3) * coordCount);
     if (colorData) {
       // cppcheck-suppress uninitvar
-      hash ^= cache::sipHash13c("colorData", 9);
-      hash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(colorData)),
+      meshHash ^= cache::sipHash13c("colorData", 9);
+      meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(colorData)),
                                 sizeof(glm::vec3) * coordCount);
     }
 
-    const cache::Key key(hash);
+    const cache::Key key(meshHash);
 
     StaticMesh *mesh;
     if (StaticMesh::createOrRetrieveFromCache(&mesh, key))
@@ -1521,16 +1521,16 @@ namespace wren {
   }
 
   StaticMesh *StaticMesh::createPointSet(int coordCount, const float *coordData, const float *colorData) {
-    uint64_t hash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
+    uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
                                       sizeof(glm::vec3) * coordCount);
     if (colorData) {
       // cppcheck-suppress uninitvar
-      hash ^= cache::sipHash13c("colorData", 9);
-      hash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(colorData)),
+      meshHash ^= cache::sipHash13c("colorData", 9);
+      meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(colorData)),
                                 sizeof(glm::vec3) * coordCount);
     }
 
-    const cache::Key key(hash);
+    const cache::Key key(meshHash);
 
     StaticMesh *mesh;
     if (StaticMesh::createOrRetrieveFromCache(&mesh, key))
@@ -1566,29 +1566,29 @@ namespace wren {
   StaticMesh *StaticMesh::createTriangleMesh(int coordCount, int indexCount, const float *coordData, const float *normalData,
                                              const float *texCoordData, const float *unwrappedTexCoordData,
                                              const unsigned int *indexData, bool outline) {
-    uint64_t hash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
+    uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
                                       sizeof(glm::vec3) * coordCount);
     // cppcheck-suppress uninitvar
-    hash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(normalData)),
+    meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(normalData)),
                               sizeof(glm::vec3) * coordCount);
     uint64_t texCoordDataHash = 0;
     if (texCoordData) {
       texCoordDataHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(texCoordData)),
                                            sizeof(glm::vec2) * coordCount);
-      hash ^= texCoordDataHash;
+      meshHash ^= texCoordDataHash;
     }
     if (unwrappedTexCoordData) {
       uint64_t unwrappedTexCoordDataHash = cache::sipHash13c(
         reinterpret_cast<const char *>(reinterpret_cast<const void *>(unwrappedTexCoordData)), sizeof(glm::vec2) * coordCount);
       if (!texCoordData || (texCoordDataHash != unwrappedTexCoordDataHash))
-        hash ^= unwrappedTexCoordDataHash;
+        meshHash ^= unwrappedTexCoordDataHash;
     }
-    hash ^= cache::sipHash13c(reinterpret_cast<const char *>(indexData), sizeof(unsigned int) * indexCount);
+    meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(indexData), sizeof(unsigned int) * indexCount);
 
     if (outline)
-      hash = ~hash;
+      meshHash = ~meshHash;
 
-    const cache::Key key(hash);
+    const cache::Key key(meshHash);
 
     StaticMesh *mesh;
     if (StaticMesh::createOrRetrieveFromCache(&mesh, key))

--- a/src/wren/StaticMesh.cpp
+++ b/src/wren/StaticMesh.cpp
@@ -537,7 +537,7 @@ namespace wren {
     uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(&params[0]), params.size() * sizeof(float));
     // cppcheck-suppress uninitvar
     meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(heightData)),
-                              sizeof(float) * dimensionX * dimensionZ);
+                                  sizeof(float) * dimensionX * dimensionZ);
     const cache::Key key(meshHash);
 
     StaticMesh *mesh;
@@ -1479,12 +1479,12 @@ namespace wren {
 
   StaticMesh *StaticMesh::createLineSet(int coordCount, const float *coordData, const float *colorData) {
     uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
-                                      sizeof(glm::vec3) * coordCount);
+                                          sizeof(glm::vec3) * coordCount);
     if (colorData) {
       // cppcheck-suppress uninitvar
       meshHash ^= cache::sipHash13c("colorData", 9);
       meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(colorData)),
-                                sizeof(glm::vec3) * coordCount);
+                                    sizeof(glm::vec3) * coordCount);
     }
 
     const cache::Key key(meshHash);
@@ -1522,12 +1522,12 @@ namespace wren {
 
   StaticMesh *StaticMesh::createPointSet(int coordCount, const float *coordData, const float *colorData) {
     uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
-                                      sizeof(glm::vec3) * coordCount);
+                                          sizeof(glm::vec3) * coordCount);
     if (colorData) {
       // cppcheck-suppress uninitvar
       meshHash ^= cache::sipHash13c("colorData", 9);
       meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(colorData)),
-                                sizeof(glm::vec3) * coordCount);
+                                    sizeof(glm::vec3) * coordCount);
     }
 
     const cache::Key key(meshHash);
@@ -1567,10 +1567,10 @@ namespace wren {
                                              const float *texCoordData, const float *unwrappedTexCoordData,
                                              const unsigned int *indexData, bool outline) {
     uint64_t meshHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(coordData)),
-                                      sizeof(glm::vec3) * coordCount);
+                                          sizeof(glm::vec3) * coordCount);
     // cppcheck-suppress uninitvar
     meshHash ^= cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(normalData)),
-                              sizeof(glm::vec3) * coordCount);
+                                  sizeof(glm::vec3) * coordCount);
     uint64_t texCoordDataHash = 0;
     if (texCoordData) {
       texCoordDataHash = cache::sipHash13c(reinterpret_cast<const char *>(reinterpret_cast<const void *>(texCoordData)),

--- a/tests/sources/test_cppcheck.py
+++ b/tests/sources/test_cppcheck.py
@@ -124,7 +124,7 @@ class TestCppCheck(unittest.TestCase):
         command = self.cppcheck + ' --enable=warning,style,performance,portability --inconclusive --force -q'
         command += ' -j %s' % str(multiprocessing.cpu_count())
         command += ' --inline-suppr --suppress=invalidPointerCast --suppress=useStlAlgorithm --suppress=uninitMemberVar '
-        command += ' --suppress=noCopyConstructor  --suppress=noOperatorEq'
+        command += ' --suppress=noCopyConstructor --suppress=noOperatorEq --suppress=strdupCalled'
         # command += ' --xml '  # Uncomment this line to get more information on the errors
         command += ' --output-file=\"' + self.reportFilename + '\"'
         for include in includeDirs:
@@ -162,6 +162,7 @@ class TestCppCheck(unittest.TestCase):
         ]
         command = self.cppcheck + ' --enable=warning,style,performance,portability --inconclusive --force -q '
         command += '--inline-suppr --suppress=invalidPointerCast --suppress=useStlAlgorithm -UKROS_COMPILATION '
+        command += '--suppress=strdupCalled '
         # command += '--xml '  # Uncomment this line to get more information on the errors
         command += '--std=c++03 --output-file=\"' + self.reportFilename + '\"'
         sources = self.add_source_files(sourceDirs, skippedDirs)

